pkg/karpenter/providers/helm/helm_test.go

```diff
@@ -0,0 +1,265 @@
+						RepositoryCache:  tmp,
+					},
+				}
+				err = installer.AddRepo("https://charts.karpenter.sh", "karpenter")
+				Expect(err).To(MatchError(ContainSubstring("failed to create new chart repository: could not find protocol handler for: ")))
+			})
+		})
+	})
+
+	Context("InstallChart", func() {
+		var (
+			fakeGetter *fakes.FakeGetter
+			getters    getter.Providers
+			tmp        string
+			err        error
+		)
+		BeforeEach(func() {
+			tmp, err = os.MkdirTemp("", "helm-testing")
+			Expect(err).NotTo(HaveOccurred())
+			fakeGetter = &fakes.FakeGetter{}
+			provider := getter.Provider{
+				Schemes: []string{"http", "https"},
+				New: func(options ...getter.Option) (getter.Getter, error) {
+					return fakeGetter, nil
+				},
+			}
+			getters = append(getters, provider)
+		})
+		AfterEach(func() {
+			_ = os.RemoveAll(tmp)
+		})
+		It("can install a test chart", func() {
+			// write out repo config
+			Expect(os.WriteFile(filepath.Join(tmp, "repositories.yaml"), []byte(expectedRepositoryYaml), 0644)).To(Succeed())
+			Expect(copy.Copy(filepath.Join("testdata", "karpenter-0.4.3.tgz"), filepath.Join(tmp, "karpenter-0.4.3.tgz"))).To(Succeed())
+			Expect(copy.Copy(filepath.Join("testdata", "karpenter-index.yaml"), filepath.Join(tmp, "karpenter-index.yaml"))).To(Succeed())
+			store := storage.Init(driver.NewMemory())
+			actionConfig := &action.Configuration{
+				Releases:     store,
+				KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+				Capabilities: chartutil.DefaultCapabilities,
+				Log:          func(format string, v ...interface{}) {},
+			}
+			installer := Installer{
+				Getters: getters,
+				Settings: &cli.EnvSettings{
+					RepositoryCache:  tmp,
+					RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+					Debug:            true,
+				},
+				ActionConfig: actionConfig,
+			}
+			err = installer.InstallChart(context.Background(), "karpenter", "karpenter/karpenter", "karpenter", "0.4.3", map[string]interface{}{
+				"controller.clusterName":     "test-security-groups-labels",
+				"controller.clusterEndpoint": "https://E2AB8AEA541E5A354CBBFACE368C534D.sk1.us-west-2.eks.amazonaws.com",
+				"serviceAccount.create":      false,
+				"defaultProvisioner.create":  false,
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+		When("locate chart is unable to find the requested chart", func() {
+			It("returns an error", func() {
+				store := storage.Init(driver.NewMemory())
+				actionConfig := &action.Configuration{
+					Releases:     store,
+					KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+					Capabilities: chartutil.DefaultCapabilities,
+					Log:          func(format string, v ...interface{}) {},
+				}
+				installer := Installer{
+					Getters: getters,
+					Settings: &cli.EnvSettings{
+						RepositoryCache:  tmp,
+						RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+						Debug:            true,
+					},
+					ActionConfig: actionConfig,
+				}
```

Comment:
might be worth putting in a before each? for readabiltiy and re-use in the other tests